pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -102,23 +103,36 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		return cmdutil.UsageError(cmd, "Must pass a name of a BuildConfig or specify build name with '--from-build' flag")
 	}
 
-	name := buildName
-	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
-		isBuild = false
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
 	}
 
-	if webhooks.Provided() {
-		return RunListBuildWebHooks(f, out, cmd.Out(), name, isBuild, webhooks.String())
+	var (
+		name     = buildName
+		resource = "builds"
+	)
+
+	if len(name) == 0 && len(args[0]) > 0 {
+		mapper, _ := f.Object()
+		resource, name, err = osutil.ResolveResource("buildconfigs", args[0], mapper)
+		if err != nil {
+			return err
+		}
+		switch resource {
+		case "buildconfigs":
```

Comment:
add a `// no special handling required` comment to make it clear this isn't intended to be a fall-through case